Microsoft.Xrm.Data.PowerShell/Microsoft.Xrm.Data.PowerShell.psm1

```diff
@@ -3943,6 +3943,74 @@ function Invoke-CrmWhoAmI{
     return $result
 }
 
+function Invoke-CrmAction {
+# .ExternalHelp Microsoft.Xrm.Data.PowerShell.Help.xml
+    [OutputType([hashtable])]
+    [OutputType([Microsoft.Xrm.Sdk.OrganizationResponse], ParameterSetName="Raw")]
+    param (
+        [Microsoft.Xrm.Tooling.Connector.CrmServiceClient]
+        $conn,
+
+        [Parameter(
+            Position=1,
+            Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $Name,
+
+        [Parameter(ValueFromPipeline)]
+        [ValidateNotNullOrEmpty()]
+        [Microsoft.Xrm.Sdk.EntityReference]
+        $Target,
+
+        [Parameter(Position=2)]
+        [ValidateNotNullOrEmpty()]
+        [hashtable]
+        $Parameters,
+
+        [Parameter(ParameterSetName="Raw")]
+        [switch]
+        $Raw
+    )
+    begin
+    {
+        $conn = VerifyCrmConnectionParam $conn
+    }
+    process
+    {
+        $request = [Microsoft.Xrm.Sdk.OrganizationRequest]::new($Name)
+
+        if($Target) {
+            $request.Parameters.Add("Target", $Target) 
+        }
+
+        if($Parameters) {
+            foreach($parameter in $Parameters.GetEnumerator()) {
+                $request.Parameters.Add($parameter.Name, $parameter.Value)
+            }
+        }
+
+        try {
+            $response = $conn.Execute($request)
+        
+            if($Raw) {
+                Write-Output $response
+            } elseif ($response.Results -and $response.Results.Count -gt 0) {
+                $outputArguments = @{}
+                foreach($outputArgument in $response.Results) {
+                    $outputArguments.Add($outputArgument.Key, $outputArgument.Value)
+                }
+                Write-Output $outputArguments
+            } else {
+                Write-Output $null
+            }
+        }
+        catch {
+            Write-Error $_
```

Comment:
Are you referring to $conn.LastCrmException? I haven't yet seen a situation where invoking execute doesn't throw an exception and also sets LastCrmException to a value - in other words, LastCrmException is only set when an exception occurs. Please explain why checking that property that would be better than what's already done here.